src/DotNetty.Common/Utilities/PriorityQueue.cs

```diff
@@ -115,5 +119,21 @@ void TrickleDown(int index, T item)
             }
             this.items[index] = item;
         }
+
+        public IEnumerator<T> GetEnumerator()
+        {
+            return this.items.Take(this.count).GetEnumerator();
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return this.GetEnumerator();
+        }
+
+        public void Clear()
+        {
+            this.count = 0;
+            this.items = new T[this.capacity];
```

Comment:
I didn't even know about `Array.Clear` - I like that option better. Fewer allocations.